activesupport/lib/active_support/core_ext/module/delegation.rb

```diff
@@ -273,6 +273,8 @@ def respond_to_missing?(name, include_private = false)
       def method_missing(method, *args, &block)
         if #{target}.respond_to?(method)
           #{target}.public_send(method, *args, &block)
+        elsif #{target}.nil?
+          raise DelegationError, "\#{method} delegated to #{target}, but #{target} is nil"
```

Comment:
It merely replaces `NoMethodError` with its subclass `DelegationError` in come cases. Any code that relies on it will still work. Does this still constitute a change in behavior, then?